src/openrct2/CmdlineSprite.cpp

```diff
@@ -708,66 +697,57 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
 
         fprintf(stdout, "Building: %s\n", spriteFilePath);
 
-        size_t i;
-        json_t* sprite_description;
+        json_t sprite_description;
 
-        json_array_foreach(sprite_list, i, sprite_description)
+        for (auto& [jsonKey, jsonSprite] : jsonSprites.items())
         {
-            if (!json_is_object(sprite_description))
+            if (!jsonSprite.is_object())
             {
-                fprintf(stderr, "Error: expected object for sprite %lu\n", static_cast<unsigned long>(i));
-                json_decref(sprite_list);
+                fprintf(stderr, "Error: expected object for sprite %s\n", jsonKey.c_str());
                 return -1;
             }
 
-            json_t* path = json_object_get(sprite_description, "path");
-            if (!json_is_string(path))
+            json_t path = jsonSprite["path"];
+            if (!path.is_string())
             {
-                fprintf(stderr, "Error: no path provided for sprite %lu\n", static_cast<unsigned long>(i));
-                json_decref(sprite_list);
+                fprintf(stderr, "Error: no path provided for sprite %s\n", jsonKey.c_str());
                 return -1;
             }
+            std::string strPath = path.get<std::string>();
+
             // Get x and y offsets, if present
-            json_t* x_offset = json_object_get(sprite_description, "x_offset");
-            json_t* y_offset = json_object_get(sprite_description, "y_offset");
+            json_t x_offset = jsonSprite["x_offset"];
+            json_t y_offset = jsonSprite["y_offset"];
 
             // Get palette option, if present
             bool keep_palette = false;
-            json_t* palette = json_object_get(sprite_description, "palette");
-            if (json_is_string(palette))
+            std::string palette = Json::GetString(jsonSprite["palette"]);
+            // simonj ~ do we need to substring here? was originally strncmp
```

Comment:
I couldn't find an example of this file so I don't know what `jsonSprite["palette"]` should contain

Reply:
Take a look at `resources/g2/sprites.json`.

Reply:
An example would be [resources/g2/sprites.json](https://github.com/OpenRCT2/OpenRCT2/blob/develop/resources/g2/sprites.json)